### PR TITLE
[libgettext] update to v2 toolchains

### PR DESCRIPTION
**Pull request by planetmarshall**
Specify library name and version:  **libgettext**

Update libgettext to the v2 autotools toolchain

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by planetmarshall** [comment]
Haven't been able to get this work for the MSVC shared build - though it worked using the V1 toolchain. I can't see any meaningful difference - maybe someone familiar with libtool could take a look at it.

```
/bin/sh ../libtool  --tag=CC   --mode=link /c/j/w/prod/buildsinglereference/.conan/data/automake/1.16.5/_/_/package/3e48e69237f7f2196164383ef9dedf0f93cbf249/res/automake-1.16/compile cl -nologo  -MD -O2 -Ob2 -Wl,--export-all-symbols -liconv  -ladvapi32 -no-undefined -export-symbols-regex '^([^g]|g[^l]|gl[^w]|glw[^t]|glwt[^h]|glwth[^r]|glwthr[^e]|glwthre[^a]|glwthrea[^d]).*' -version-info 9:6:1 -rpath /lib -L/c/j/w/prod/buildsinglereference/.conan/data/libiconv/1.17/_/_/package/6cc50b139b9c3d27b3e9042d5f5372d327b3a9f7/lib -Wl,--disable-auto-import -o libgnuintl.la  bindtextdom.lo dcgettext.lo dgettext.lo gettext.lo finddomain.lo hash-string.lo loadmsgcat.lo localealias.lo textdomain.lo l10nflist.lo explodename.lo dcigettext.lo dcngettext.lo dngettext.lo ngettext.lo plural.lo plural-exp.lo localcharset.lo threadlib.lo lock.lo relocatable.lo langprefs.lo localename.lo localename-table.lo log.lo printf.lo setlocale.lo version.lo xsize.lo osdep.lo intl-compat.lo libintl.res.lo 

*** Warning: linker path does not have real file for library -liconv.
*** I have the capability to make that library automatically link in when
*** you link to this library.  But I can only do this if you have a
*** shared version of the library, which you do not appear to have
*** because I did check the linker path looking for a file starting
*** with iconv and none of the candidates passed a file format test
*** using a file magic. Last file checked: /c/J/w/prod/BuildSingleReference/.conan/data/libiconv/1.17/_/_/package/6cc50b139b9c3d27b3e9042d5f5372d327b3a9f7/bin/iconv.exe
*** The inter-library dependencies that have been dropped here will be
*** automatically added whenever a program is linked with this library
*** or is declared to -dlopen it.

*** Since this library must not contain undefined symbols,
*** because either the platform does not support them or
*** it was explicitly requested with -no-undefined,
*** libtool will only create a static version of it.
```

**Comment by planetmarshall** [comment]
> Haven't been able to get this work for the MSVC shared build - though it worked using the V1 toolchain. I can't see any meaningful difference - maybe someone familiar with libtool could take a look at it.
> 

OK, fixed (hopefully) by mimicking some of the environment variables set by the v1 toolchain that are absent from v2 (notably `_LINK_` and `LIB`, and unsetting `LIBS`)

**Review by prince-chrismc — DISMISSED** [review]

File: recipes/libgettext/all/test_v1_package/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+cmake_minimum_required(VERSION 3.1)
+project(test_package C)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
```
Comment (uilianries) [inline recipes/libgettext/all/test_v1_package/CMakeLists.txt:5]:
```suggestion
conan_basic_setup(TARGETS)
```
To enforce only cmake targets and avoid variables

**Review by prince-chrismc — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM

**Pull request merged**